SNT/TypeGenerator.cs

```diff
@@ -0,0 +1,1264 @@
+using System;
+using System.Buffers;
+using System.Linq;
+using System.Numerics.Tensors;
+using System.Runtime.CompilerServices;
+using TorchSharp;
+
+namespace Torch.SNT 
+{
+
+    /// <summary>
+    ///   Wrapper class used to surface a Torch ByteTensor as a System.Numerics DensorTensor of byte
+    /// </summary>
+    public sealed class ByteTorchTensor : DenseTensor<byte>, IDisposable
+    {
+        internal sealed class ByteNativeMemory : MemoryManager<byte>
+        {
+            private readonly ByteTensor.ByteStorage storage;
+
+            public ByteNativeMemory(ByteTensor.ByteStorage storage)
+            {
+                this.storage = storage;
+
+                if (storage.Size () < 0)
+                {
+                    throw new ArgumentOutOfRangeException ("Length cannot be negative.");
+                }
+            }
+
+            /// <summary>
+            /// Returns a span wrapping the underlying memory.
+            /// Remember to Unpin the memory once the span is disposed.
+            /// </summary>
+            public override Span<byte> GetSpan ()
+            {
+                ulong len = storage.Size ();
+
+                if (len > int.MaxValue)
+                {
+                    throw new InvalidCastException ("Tensor size not supported.");
+                }
+
+                unsafe
+                {
+                    return new Span<byte> (storage.Data.ToPointer (), (int)len);
+                }
+            }
+
+            /// <summary>
+            /// Returns a handle to the memory that has been pinned and hence its address can be taken.
+            /// </summary>
+            /// <param name="elementIndex">The offset to the element within the memory at which the returned <see cref="MemoryHandle"/> points to. (default = 0)</param>
+            public override MemoryHandle Pin (int elementIndex = 0)
+            {
+                unsafe
+                {
+                    storage.Retain();
+                    if ((uint)elementIndex > storage.Size ()) throw new ArgumentOutOfRangeException (nameof (elementIndex), "Index out of array bound.");
+                    void* pointer = Unsafe.Add<byte> ((void*)storage.Data, elementIndex);
+                    return new MemoryHandle (pointer, default, this);
+                }
+            }
+
+            /// <summary>
+            /// Lets the garbage collector know that the object is free to be moved now.
+            /// </summary>
+            public override void Unpin ()
+            {
+                storage.Free ();
+            }
+
+            protected override void Dispose (bool disposing)
+            {
+                if (disposing)
+                {
+                    storage.Free ();
+                }
+            }
+        }
+
+        private readonly object inner;
```

Comment:
Why is this field held as `object`? why not `ByteTensor`?

Reply:
This is something coming from the previous version using generics. Fixed.